Source/Editor/Viewport/MainEditorGizmoViewport.cs

```diff
@@ -576,15 +610,202 @@ protected override void OrientViewport(ref Quaternion orientation)
                 base.OrientViewport(ref orientation);
         }
 
+        /// <inheritdoc />
+        public override void OnMouseMove(Float2 location)
+        {
+            base.OnMouseMove(location);
+
+            // Dont allow rubber band selection when gizmo is controlling mouse, vertex painting mode, or cloth painting is enabled
+            if (_isRubberBandSpanning && ((Gizmos.Active.IsControllingMouse || Gizmos.Active is VertexPaintingGizmo || Gizmos.Active is ClothPaintingGizmo) || IsControllingMouse || IsRightMouseButtonDown))
```

Comment:
I would move whole logic from this function (and some local state vars fro the top) into a separate class like `RubberBandSelector` to be reusable in other viewports (like prefab in future) and have more organized code.

Reply:
Done